libyul/backends/evm/EVMDialect.cpp

```diff
@@ -366,6 +394,76 @@ std::map<YulName, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _ev
 					_assembly.appendAuxDataLoadN(static_cast<uint16_t>(literal->value.value()));
 				}
 			));
+
+			builtins.emplace(createFunction(
+				"eofcreate",
+				5,
+				1,
+				SideEffects{
+					false,               // movable
+					false,               // movableApartFromEffects
+					false,               // canBeRemoved
+					false,               // canBeRemovedIfNotMSize
+					true,                // cannotLoop
+					SideEffects::Write,  // otherState
+					SideEffects::Write,  // storage
+					SideEffects::Write,  // memory
+					SideEffects::Write   // transientStorage
+				},
+				{LiteralKind::String, std::nullopt, std::nullopt, std::nullopt, std::nullopt},
+				[](
+					FunctionCall const& _call,
+					AbstractAssembly& _assembly,
+					BuiltinContext& context
+				) {
+					yulAssert(_call.arguments.size() == 5, "");
+					Literal const* literal = std::get_if<Literal>(&_call.arguments.front());
+					yulAssert(literal, "");
+					auto const it = context.subIDs.find(formatLiteral(*literal));
+					yulAssert(it != context.subIDs.end(), "");
+					yulAssert((*it).second <= std::numeric_limits<solidity::yul::AbstractAssembly::ContainerID>::max(), "");
+					_assembly.appendEofCreate(static_cast<solidity::yul::AbstractAssembly::ContainerID>((*it).second));
+				}
+				));
+
+			{
+				auto [createdFunctionIt, success] = builtins.emplace(createFunction(
+					"returncontract",
+					3,
+					0,
+					SideEffects{
+						false,               // movable
+						false,               // movableApartFromEffects
+						false,               // canBeRemoved
+						false,               // canBeRemovedIfNotMSize
+						true,                // cannotLoop
+						SideEffects::None,   // otherState
+						SideEffects::None,   // storage
+						SideEffects::Write,  // memory
+						SideEffects::None    // transientStorage
+					},
+					{LiteralKind::String, std::nullopt, std::nullopt},
+					[](
+						FunctionCall const& _call,
+						AbstractAssembly& _assembly,
+						BuiltinContext& context
+					) {
+						yulAssert(_call.arguments.size() == 3, "");
+						Literal const* literal = std::get_if<Literal>(&_call.arguments.front());
+						yulAssert(literal, "");
+						auto const it = context.subIDs.find(formatLiteral(*literal));
+						yulAssert(it != context.subIDs.end(), "");
+						yulAssert((*it).second <= std::numeric_limits<solidity::yul::AbstractAssembly::ContainerID>::max(), "");
+						_assembly.appendReturnContract(static_cast<solidity::yul::AbstractAssembly::ContainerID>((*it).second));
```

Comment:
This could be simplified a bit:
- The message in asserts is optional.
- We're in `solidity::yul` namespace here.
- I'd avoid micro-optimizing `map` lookups or `variant` casts, at least not in non-performance critical code. They usually makes things hard to read and is not worth it IMO.
- We also have `valueOrDefault()`, `valueOrNull()`, which should actually make it just as efficient as with iterators.

```suggestion
						yulAssert(_call.arguments.size() == 3);
						yulAssert(std::holds_alternative<Literal>(_call.arguments.front()));
						std::string subObjectName = formatLiteral(std::get<Literal>(_call.arguments.front()));
						AbstractAssembly::SubID* containerID = util::valueOrNull(	context.subIDs,	subObjectName);
						yulAssert(containerID && *containerID <= std::numeric_limits<AbstractAssembly::ContainerID>::max());
						_assembly.appendReturnContract(static_cast<AbstractAssembly::ContainerID>(*containerID);
```

Reply:
Changed.